### ISSUE TITLE
`Track index out of range` Error When Initial MKVTrack

### ISSUE THREAD

**Issue by Masterain98** [GitBib/pymkv2#46]
### **Describe the bug**
When adapting pymkv2 from pymkv library, I notice that when initialing an MKVTrack from an existing file, an `IndexError: track index out of range` error will occur.

### **To Reproduce**
My source code: 
https://github.com/Masterain98/PlexMuxy/blob/a0e9c9fab409667a3daaf7f3a34ad7befaa1b617/main.py#L107-L115

---
Minimum reproduce:

```
    mkv = pymkv.MKVFile("./sample.mkv", mkvmerge_path=MKVMerge)
    mkv.add_track(pymkv.MKVTrack("./sample.JPSC.ass", mkvmerge_path=MKVMerge))
```

### **Expected behavior**
MKVTrack can be created, separated, and added to MKVFile. This can be achieved in the old pymkv library without any error.

### **Screenshots**

```python
Traceback (most recent call last):
  File "C:\Users\i\Documents\GitHub\pymkv2\simple_tests\simple_test.py", line 11, in <module>
    process_file()
  File "C:\Users\i\Documents\GitHub\pymkv2\simple_tests\simple_test.py", line 8, in process_file
    mkv.add_track(pymkv.MKVTrack("./sample.JPSC.ass", mkvmerge_path=MKVMerge))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\i\Documents\GitHub\pymkv2\pymkv\MKVTrack.py", line 145, in __init__
    self.file_path = file_path
    ^^^^^^^^^^^^^^
  File "C:\Users\i\Documents\GitHub\pymkv2\pymkv\MKVTrack.py", line 218, in file_path
    self.track_id = 0
    ^^^^^^^^^^^^^
  File "C:\Users\i\Documents\GitHub\pymkv2\pymkv\utils.py", line 98, in wrapper
    return method(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\i\Documents\GitHub\pymkv2\pymkv\MKVTrack.py", line 289, in track_id
    raise IndexError(msg)
IndexError: track index out of range
```

### **Software (please complete the following information):**
 - OS: Windows 10 22H2
 - MKVToolNix v88

**Comment by Masterain98** [comment]
Looks like the issue comes from here: https://github.com/GitBib/pymkv2/blob/51f73ea420a9b8ee4f03f54425c0483422dda9f9/pymkv/MKVTrack.py#L286-L287

For new initialed MKVTrack, `len(tracks)` will always be zero, because `_info_json` is `{}`

https://github.com/GitBib/pymkv2/blob/51f73ea420a9b8ee4f03f54425c0483422dda9f9/pymkv/MKVTrack.py#L143

**Comment by GitBib** [comment]
The `MKVTrack` class is intended as an internal structure within `MKVFile`. If you need to create a file, you can do so through `MKVFile` and then use `add_file`.

**Comment by GitBib** [comment]
However, I suspect that the ensure_info decorator is not performing its intended function. I will verify this.

**Comment by GitBib** [comment]
Fixed the issue in #48 . Now, creation via a track should work without any problems.

**Comment by Masterain98** [comment]
Thank you @GitBib, I tested on my side with the latest version, and it works. 

---

An off-topic question: You marked the `is_bcp47` function as deprecated. What's the plan for this module? In my project, I handled some subtitle files with very specific regional codes and got errors (highfestiva/bcp47.py#7), so I'm trying make a new verification method.
